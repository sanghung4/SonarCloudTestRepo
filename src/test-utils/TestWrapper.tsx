import React, { JSXElementConstructor, ReactElement, ReactNode } from 'react';

import { ThemeProvider } from '@dialexa/reece-component-library';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import { OktaAuth } from '@okta/okta-auth-js';
import { render as jestRender, RenderResult } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';

import { UserContextType, AuthContext } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';
import i18n from 'test-utils/mockI18N';
import {
  SelectedAccountsContext,
  SelectedAccountsContextType
} from 'providers/SelectedAccountsProvider';
import { CartContext, CartContextType } from 'providers/CartProvider';
import {
  CheckoutContext,
  CheckoutContextType,
  initialCheckoutContextValues
} from 'Checkout/CheckoutProvider';

/**
 * NOTE:
 * To setup mocks follow the examples here:
 * https://www.apollographql.com/docs/react/development-testing/testing/
 *
 * To get a successful response in your test (not just the loading state)
 * you can do the following:
 *
 * it('Should render', async () => {
 *   const { container } = renderMockRouter(<MyComponent />, mocks);
 *
 *   await waitFor(() => new Promise((res) => setTimeout(res, 0)));
 *
 *   expect(container).toMatchSnapshot();
 * });
 *
 */

/******************************/
/* Types                      */
/******************************/
type Props = {
  children?: ReactNode;
  authConfig?: Partial<UserContextType>;
  selectedAccountsConfig?: Partial<SelectedAccountsContextType>;
  cartConfig?: Partial<CartContextType>;
  checkoutConfig?: Partial<CheckoutContextType>;
  mocks?: ReadonlyArray<MockedResponse>;
};

type WrapperProps<TKey extends keyof Props> = {
  value: Props[TKey];
  children: React.ReactNode;
};

/******************************/
/* Defaults                   */
/******************************/
const defaultAuth: UserContextType = {
  authState: {
    isAuthenticated: true
  },
  isLoggingOut: false,
  oktaAuth: new OktaAuth({ issuer: 'http://test.com' }),
  profile: {
    userId: 'testuser',
    permissions: Object.values(Permission),
    isEmployee: false,
    isVerified: false
  }
};

const defaultSelectedAccounts: SelectedAccountsContextType = {
  clearAccounts: () => {},
  updateAccounts: () => {},
  updateShippingBranchId: () => {},
  loading: false,
  selectedAccounts: {
    billTo: { erpAccountId: '' },
    shipTo: { erpAccountId: '' },
    shippingBranchId: ''
  },
  isMincron: false,
  isEclipse: false
};

const defaultCartContextValue: CartContextType = {
  addItemToCart: () => {},
  addItemsToCart: () => {},
  addAllListItemsToCart: () => {},
  cart: undefined,
  cartLoading: false,
  checkingOutWithQuote: false,
  contractBranch: undefined,
  clearContract: () => {},
  clearQuote: () => {},
  deleteCartItems: () => new Promise(() => {}),
  disableAddToCart: false,
  deleteItem: () => {},
  getUserCart: () => new Promise(() => {}),
  itemCount: 0,
  itemLoading: undefined,
  lineNotes: {},
  releaseContractToCart: () => {},
  setContract: () => {},
  setLineNotes: () => {},
  setQuoteId: () => {},
  setQuoteData: () => {},
  setQuoteShipToId: () => {},
  setPreviousCart: () => {},
  setSelectedBranch: () => {},
  updateCart: undefined,
  updateDelivery: undefined,
  updateItemQuantity: () => {},
  updateWillCall: undefined,
  updateWillCallBranch: undefined,
  refreshCart: undefined,
  isWillCall: false,
  deleteCreditCardFromCart: () => new Promise(() => {})
};

/******************************/
/* Wrappers                   */
/******************************/
function SelectedAccountsProviderWrapper({
  value,
  children
}: WrapperProps<'selectedAccountsConfig'>) {
  if (value) {
    return (
      <SelectedAccountsContext.Provider
        value={{ ...defaultSelectedAccounts, ...value }}
      >
        {children}
      </SelectedAccountsContext.Provider>
    );
  }

  return <>{children}</>;
}

function CartProviderWrapper({ value, children }: WrapperProps<'cartConfig'>) {
  if (value) {
    return (
      <CartContext.Provider value={{ ...defaultCartContextValue, ...value }}>
        {children}
      </CartContext.Provider>
    );
  }

  return <>{children}</>;
}

function CheckoutProviderWrapper({
  value,
  children
}: WrapperProps<'checkoutConfig'>) {
  if (value) {
    return (
      <CheckoutContext.Provider
        value={{ ...initialCheckoutContextValues, ...value }}
      >
        {children}
      </CheckoutContext.Provider>
    );
  }

  return <>{children}</>;
}

function TestWrapper(props: Props) {
  return (
    <ThemeProvider>
      <I18nextProvider i18n={i18n}>
        <AuthContext.Provider value={{ ...defaultAuth, ...props.authConfig }}>
          <MockedProvider mocks={props.mocks ?? []}>
            <SelectedAccountsProviderWrapper
              value={props.selectedAccountsConfig}
            >
              <CartProviderWrapper value={props.cartConfig}>
                <CheckoutProviderWrapper value={props.checkoutConfig}>
                  {props.children}
                </CheckoutProviderWrapper>
              </CartProviderWrapper>
            </SelectedAccountsProviderWrapper>
          </MockedProvider>
        </AuthContext.Provider>
      </I18nextProvider>
    </ThemeProvider>
  );
}

function jestRenderFn(wrapped: JSX.Element) {
  return jestRender(wrapped, {
    wrapper: MemoryRouter as
      | JSXElementConstructor<{
          children: ReactElement<any, string | JSXElementConstructor<any>>;
        }>
      | undefined
  });
}

export function render(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return jestRenderFn(wrapped);
}
export type TestRender = RenderResult<
  typeof import('@testing-library/dom/types/queries'),
  Element | DocumentFragment
>;

export function renderWithComponent(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return {
    render: jestRenderFn(wrapped),
    component: wrapped
  };
}

export default TestWrapper;
