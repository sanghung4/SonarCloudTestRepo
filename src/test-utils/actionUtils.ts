import { fireEvent, findByTestId, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

export const fillFormTextInput = async (testId: string, value: string) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, `${testId}-input`);
    await fireEvent.change(textInput, { target: { value } });
  });
};

export const fillMaskedFormTextInput = async (
  testId: string,
  value: string
) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, `${testId}-input`);
    await userEvent.type(textInput, value);
  });
};

export const clickButton = async (testId: string) => {
  await act(async () => {
    const button = await findByTestId(document.body, testId);
    await fireEvent.click(button);
  });
};

export const clickRadio = async (testId: string, optionIndex: number) => {
  await act(async () => {
    const radio = await findByTestId(
      document.body,
      `${testId}-radio-${optionIndex}`
    );
    await userEvent.click(radio);
  });
};

export const fillFormSelectInput = async (testId: string, value: string) => {
  await act(async () => {
    const selectInput = await findByTestId(document.body, `${testId}-input`);
    await fireEvent.change(selectInput, { target: { value } });
    await fireEvent.keyDown(window, { key: 'Enter' });
  });
};

export const focusInput = async (testId: string) => {
  await act(async () => {
    const input = await findByTestId(document.body, testId);
    await fireEvent.focus(input);
  });
};

export const changeInput = async (testId: string, value: string) => {
  await act(async () => {
    const textInput = await findByTestId(document.body, testId);
    await fireEvent.change(textInput, { target: { value } });
  });
};

export const blurInput = async (testId: string) => {
  await act(async () => {
    const input = await findByTestId(document.body, testId);
    await fireEvent.blur(input);
  });
};

export const keyboardDown = async (
  key: string,
  element?: Element | Node | Document | Window
) => {
  await act(async () => {
    await fireEvent.keyDown(element ?? window, { key });
  });
};
