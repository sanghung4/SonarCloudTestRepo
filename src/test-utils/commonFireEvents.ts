import { act, fireEvent } from '@testing-library/react';

export function actClickOnElement(element: HTMLElement) {
  act(() => {
    fireEvent.click(element);
  });
}
export function actUpdateInput(input: HTMLElement, value: string) {
  act(() => updateInput(input, value));
}
export function updateInput(input: HTMLElement, value: string) {
  fireEvent.focus(input);
  fireEvent.change(input, { target: { value } });
  fireEvent.blur(input);
}
