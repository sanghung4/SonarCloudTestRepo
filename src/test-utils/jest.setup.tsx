import React, { JSXElementConstructor, ReactElement, ReactNode } from "react";
import { MockedProvider, MockedResponse } from "@apollo/client/testing";
import { MemoryRouter } from "react-router-dom";
import { render as jestRender } from "@testing-library/react";

export const mockApollo = jest.mock(
  "@apollo/client/testing",
  () => MockedProvider
);

interface Props {
  children?: ReactNode;
  mocks?: ReadonlyArray<MockedResponse>;
}

function TestWrapper(props: Props) {
  return (
    <MockedProvider mocks={props.mocks ?? []}>{props.children}</MockedProvider>
  );
}

export function render(component: ReactNode, config?: Props) {
  const wrapped = <TestWrapper {...config}>{component}</TestWrapper>;
  return jestRender(wrapped, {
    wrapper: MemoryRouter as
      | JSXElementConstructor<{
          children: ReactElement<any, string | JSXElementConstructor<any>>;
        }>
      | undefined,
  });
}
