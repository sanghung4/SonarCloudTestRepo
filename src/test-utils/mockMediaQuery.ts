import mediaQuery from 'css-mediaquery';

function createMatchMedia(width: number) {
  return (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: () => jest.fn(),
    removeListener: () => jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  });
}

export const breakpoints = {
  mobile: 414,
  desktop: 1200
};

export const setBreakpoint = (breakpoint: 'mobile' | 'desktop' = 'desktop') => {
  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: breakpoints[breakpoint]
  });
  window.dispatchEvent(new Event('resize'));

  window.matchMedia = createMatchMedia(window.innerWidth);
};
