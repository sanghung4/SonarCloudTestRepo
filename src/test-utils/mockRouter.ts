import { useHistory, useLocation } from 'react-router-dom';

export const mockHistory: ReturnType<typeof useHistory> = {
  length: 0,
  action: 'PUSH',
  location: {
    pathname: '',
    search: '',
    state: '',
    hash: ''
  },
  push: jest.fn(),
  replace: jest.fn(),
  go: jest.fn(),
  goBack: jest.fn(),
  goForward: jest.fn(),
  block: jest.fn(),
  listen: jest.fn(),
  createHref: jest.fn()
};

export const mockLocation: ReturnType<typeof useLocation> = {
  pathname: '',
  search: '',
  state: undefined,
  hash: ''
};
