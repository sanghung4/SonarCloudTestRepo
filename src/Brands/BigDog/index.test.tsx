import BigDogMarketing from 'Brands/BigDog/index';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';

describe('Brands - BigDog', () => {
  it('index should match the snapshot on desktop', () => {
    setBreakpoint('desktop');
    const { container } = render(<BigDogMarketing />);
    expect(container).toMatchSnapshot();
  });
  it('index should match the snapshot on mobile', () => {
    setBreakpoint('mobile');
    const { container } = render(<BigDogMarketing />);
    expect(container).toMatchSnapshot();
  });
});
