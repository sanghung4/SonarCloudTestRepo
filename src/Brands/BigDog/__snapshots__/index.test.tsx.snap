// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brands - BigDog index should match the snapshot on desktop 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-okf7m7-MuiContainer-root"
    style="max-width: 1440px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-hxgllp-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-90i9l4"
      >
        <div
          class="MuiGrid-root css-o8xg47-MuiGrid-root"
        >
          <svg
            style="width: 238px; height: 50px;"
          >
            BigDogLogo.svg
          </svg>
          <div
            class="MuiGrid-root css-vj1n65-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1oobtzp-MuiTypography-root"
            >
              RULE THE WORKSITE.
            </p>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-sa15y2-MuiDivider-root"
      />
    </div>
    <div
      class="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-i72534-MuiTypography-root"
      >
        We've been redesigning and improving the BIGDOG range to meet the exact needs of contractors and technicians.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-120lkjs-MuiTypography-root"
      >
        - so you get even more bang for your bark.
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-5hwgfl-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-l22kh7-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-15cryil"
        >
          <img
            class="MuiBox-root css-0"
            src="image1.jpg"
          />
        </div>
        <div
          class="MuiGrid-root css-1755usu-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1gxt4hz"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-dfxjru-MuiTypography-root"
            >
              Smarter Designs
            </p>
          </div>
          <div
            class="MuiBox-root css-2zvt8e"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1vlrsf3-MuiTypography-root"
            >
              We're continuously looking to integrate clever little features into all our tools so you can work faster.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-1p73i44-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-15cryil"
        >
          <img
            class="MuiBox-root css-0"
            src="image2.jpg"
          />
        </div>
        <div
          class="MuiGrid-root css-1755usu-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1gxt4hz"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-dfxjru-MuiTypography-root"
            >
              Heavy-Duty Quality
            </p>
          </div>
          <div
            class="MuiBox-root css-2zvt8e"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1vlrsf3-MuiTypography-root"
            >
              Our latest designs are made to last, using commercial grade materials that make light work of even the toughest plumbing jobs.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-1hrtl0j-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-15cryil"
        >
          <img
            class="MuiBox-root css-0"
            src="image3.jpg"
          />
        </div>
        <div
          class="MuiGrid-root css-1755usu-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1gxt4hz"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-dfxjru-MuiTypography-root"
            >
              Always On-Hand
            </p>
          </div>
          <div
            class="MuiBox-root css-2zvt8e"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1vlrsf3-MuiTypography-root"
            >
              BIGDOG is never far away. Youâ€™ll find them at the local Reece branch. Plus, you can order through maX and have them delivered to site straight away.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root css-smc5z0-MuiGrid-root"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-4mzf4j-MuiDivider-root"
      />
      <div
        class="MuiGrid-root css-zzurbn-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-xrqeot-MuiTypography-root"
        >
          FEATURED ITEMS
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item css-q1swar-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-16g8vvv-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585329"
          >
            <div
              class="MuiBox-root css-or4m32"
            >
              <img
                alt="26 ft Magnetic Tape Measure"
                src="BigdogMeasuringTape.jpg"
              />
            </div>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585329"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-yigre9-MuiTypography-root"
            >
              26 ft Magnetic Tape Measure
            </p>
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-16g8vvv-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585382"
          >
            <div
              class="MuiBox-root css-or4m32"
            >
              <img
                alt="Round Point Shovel"
                src="BigdogPointShovel.jpg"
              />
            </div>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585382"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-yigre9-MuiTypography-root"
            >
              Round Point Shovel
            </p>
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-16g8vvv-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585442"
          >
            <div
              class="MuiBox-root css-or4m32"
            >
              <img
                alt="Torpedo Level"
                src="BigdogTorpedoLevel.jpg"
              />
            </div>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585442"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-yigre9-MuiTypography-root"
            >
              Torpedo Level
            </p>
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-16g8vvv-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585403"
          >
            <div
              class="MuiBox-root css-or4m32"
            >
              <img
                alt="Multi Purpose Tool Pouch"
                src="BigdogToolPouch.jpg"
              />
            </div>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585403"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-yigre9-MuiTypography-root"
            >
              Multi Purpose Tool Pouch
            </p>
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-16g8vvv-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585395"
          >
            <div
              class="MuiBox-root css-or4m32"
            >
              <img
                alt="Ratchet Tie Down Strap"
                src="BigdogTieDownStrap.jpg"
              />
            </div>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585395"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-yigre9-MuiTypography-root"
            >
              Ratchet Tie Down Strap
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root css-da5kf6-MuiGrid-root"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-sa15y2-MuiDivider-root"
      />
      <div
        class="MuiGrid-root css-dttsxh-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-krzudk-MuiTypography-root"
        >
          SEE THE FULL RANGE.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-a3ady9-MuiTypography-root"
        >
          With more than 40 products in the BIGDOG range, weâ€™ve got a tool for every application.
        </p>
        <div
          class="MuiGrid-root MuiGrid-container css-5qo8ve-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/search?filters=brand%7CALLPROFESSIONAL+MFG+CO+LTD"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-i32lnv-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-zp579f-MuiTypography-root"
              >
                See the Range
              </p>
            </button>
          </a>
        </div>
      </div>
      <img
        class="MuiBox-root css-it8fua"
        src="footerImageDesktop.jpg"
      />
    </div>
  </div>
</div>
`;

exports[`Brands - BigDog index should match the snapshot on mobile 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-okf7m7-MuiContainer-root"
    style="max-width: 1440px;"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-hxgllp-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-90i9l4"
      >
        <div
          class="MuiGrid-root css-o8xg47-MuiGrid-root"
        >
          <svg
            style="width: 160px; height: 34px;"
          >
            BigDogLogo.svg
          </svg>
          <div
            class="MuiGrid-root css-vj1n65-MuiGrid-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1oobtzp-MuiTypography-root"
            >
              RULE THE WORKSITE.
            </p>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-sa15y2-MuiDivider-root"
      />
    </div>
    <div
      class="MuiGrid-root css-vj1n65-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-zcopyv-MuiTypography-root"
      >
        We've been redesigning and improving the BIGDOG range to meet the exact needs of contractors and technicians.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-mwstep-MuiTypography-root"
      >
        - so you get even more bang for your bark.
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container css-177e0ig-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-22jcvu-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-15cryil"
        >
          <img
            class="MuiBox-root css-0"
            src="image1.jpg"
          />
        </div>
        <div
          class="MuiGrid-root css-1itlxog-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-934uvy"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-dfxjru-MuiTypography-root"
            >
              Smarter Designs
            </p>
          </div>
          <div
            class="MuiBox-root css-2zvt8e"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1vlrsf3-MuiTypography-root"
            >
              We're continuously looking to integrate clever little features into all our tools so you can work faster.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-180hvbr-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-15cryil"
        >
          <img
            class="MuiBox-root css-0"
            src="image2.jpg"
          />
        </div>
        <div
          class="MuiGrid-root css-1itlxog-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-934uvy"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-dfxjru-MuiTypography-root"
            >
              Heavy-Duty Quality
            </p>
          </div>
          <div
            class="MuiBox-root css-2zvt8e"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1vlrsf3-MuiTypography-root"
            >
              Our latest designs are made to last, using commercial grade materials that make light work of even the toughest plumbing jobs.
            </p>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-1rlent-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-15cryil"
        >
          <img
            class="MuiBox-root css-0"
            src="image3.jpg"
          />
        </div>
        <div
          class="MuiGrid-root css-1itlxog-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-934uvy"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-dfxjru-MuiTypography-root"
            >
              Always On-Hand
            </p>
          </div>
          <div
            class="MuiBox-root css-2zvt8e"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1vlrsf3-MuiTypography-root"
            >
              BIGDOG is never far away. Youâ€™ll find them at the local Reece branch. Plus, you can order through maX and have them delivered to site straight away.
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root css-a7e5e4-MuiGrid-root"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-4mzf4j-MuiDivider-root"
      />
      <div
        class="MuiGrid-root css-6o7512-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1c6t02i-MuiTypography-root"
        >
          FEATURED ITEMS
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item css-7y7a7j-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-jtoaah-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585329"
          >
            <div
              class="MuiBox-root css-tkevr6"
            >
              <img
                alt="26 ft Magnetic Tape Measure"
                src="BigdogMeasuringTape.jpg"
              />
            </div>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585329"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-yigre9-MuiTypography-root"
            >
              26 ft Magnetic Tape Measure
            </p>
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-jtoaah-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585382"
          >
            <div
              class="MuiBox-root css-tkevr6"
            >
              <img
                alt="Round Point Shovel"
                src="BigdogPointShovel.jpg"
              />
            </div>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585382"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-yigre9-MuiTypography-root"
            >
              Round Point Shovel
            </p>
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-jtoaah-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585442"
          >
            <div
              class="MuiBox-root css-tkevr6"
            >
              <img
                alt="Torpedo Level"
                src="BigdogTorpedoLevel.jpg"
              />
            </div>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585442"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-yigre9-MuiTypography-root"
            >
              Torpedo Level
            </p>
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-jtoaah-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585403"
          >
            <div
              class="MuiBox-root css-tkevr6"
            >
              <img
                alt="Multi Purpose Tool Pouch"
                src="BigdogToolPouch.jpg"
              />
            </div>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585403"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-yigre9-MuiTypography-root"
            >
              Multi Purpose Tool Pouch
            </p>
          </a>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-jtoaah-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585395"
          >
            <div
              class="MuiBox-root css-tkevr6"
            >
              <img
                alt="Ratchet Tie Down Strap"
                src="BigdogTieDownStrap.jpg"
              />
            </div>
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/product/MSC-1585395"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-yigre9-MuiTypography-root"
            >
              Ratchet Tie Down Strap
            </p>
          </a>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root css-268fxx-MuiGrid-root"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-sa15y2-MuiDivider-root"
      />
      <div
        class="MuiGrid-root css-qdn0gb-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1xuh5pi-MuiTypography-root"
        >
          SEE THE FULL RANGE.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1fqoe6d-MuiTypography-root"
        >
          With more than 40 products in the BIGDOG range, weâ€™ve got a tool for every application.
        </p>
        <div
          class="MuiGrid-root MuiGrid-container css-35eu0k-MuiGrid-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-fe2j4-MuiTypography-root-MuiLink-root"
            href="/search?filters=brand%7CALLPROFESSIONAL+MFG+CO+LTD"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-i32lnv-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-zp579f-MuiTypography-root"
              >
                See the Range
              </p>
            </button>
          </a>
        </div>
      </div>
      <img
        class="MuiBox-root css-vy99a4"
        src="footerImageMobile.jpg"
      />
    </div>
  </div>
</div>
`;
